internal/migrations/upgrade_repo_v2.go

```diff
@@ -0,0 +1,121 @@
+package migrations
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"path/filepath"
+
+	"github.com/restic/restic/internal/restic"
+)
+
+func init() {
+	register(&UpgradeRepoV2{})
+}
+
+type UpgradeRepoV2Error struct {
+	UploadNewConfigError   error
+	ReuploadOldConfigError error
+
+	BackupFilePath string
+}
+
+func (err UpgradeRepoV2Error) Error() string {
```

Comment:
I suggest using pointer receivers on this type, and in fact, on all custom error types. That avoids the question "do I check for UpgradeRepoV2Error, or *UpgradeRepoV2Error"? The stdlib does this for all but some of the oldest error types.

(Not a big deal here, but it sets an example for other contributors.)

Reply:
Is changed. I've noticed that using the pointer receiver the compiler actually enforces the use of the pointer type when checking for a specific type. So it's both consistent and cannot be used the wrong way :) .